distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1543,6 +1543,196 @@ public void testFederatedDownloadProductToCacheOnlyCacheDisabled() throws Except
+
+        validateNotificationsFailure(cometDClient1, filename);
+        validateNotificationsFailure(cometDClient2, filename);
+
+        validateActivitiesFailure(cometDClient1, filename);
+        validateActivitiesFailure(cometDClient2, filename);
+
+    }
+
+    private void setupCswServer(String metacardId, String filename, String resourceData)
+            throws Exception {
+        cswServer.whenHttp()
+                .match(post("/services/csw"),
+                        withPostBodyContaining("GetRecords"),
+                        withPostBodyContaining(metacardId))
+                .then(ok(),
+                        contentType("text/xml"),
+                        bytesContent(getCswQueryResponse(metacardId).getBytes()));
+
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"),
+                        Condition.parameter("id", metacardId))
+                .then(getCswRetrievalHeaders(filename),
+                        chunkedContentWithHeaders(resourceData, Duration.ofMillis(200), 3));
+
+    }
+
+    private void checkExpectations(CometDClient cometDClient, Integer numNotifications,
+            Integer numActivities) throws Exception {
+        expect("Waiting for notifications").within(10, SECONDS)
+                .until(() -> cometDClient.getMessages(NOTIFICATIONS_CHANNEL)
+                        .size() == numNotifications);
+        expect("Waiting for activities").within(10, SECONDS)
+                .until(() -> cometDClient.getMessages(ACTIVITIES_CHANNEL)
+                        .size() == numActivities);
+
+    }
+
+    private void validateNotificationsSuccess(CometDClient cometDClient, String filename,
+            String resourceData) throws Exception {
+
+        List<String> notifications = cometDClient.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
+        assertThat(notifications.size(), is(1));
+        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(0)),
+                filename,
+                getResourceRetrievalCompletedMessage(resourceData.length()),
+                "complete");
+
+    }
+
+    private void validateActivitiesSuccess(CometDClient cometDClient, String filename,
+            String resourceData) throws Exception {
+        List<String> notifications = cometDClient.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
+        assertThat(notifications.size(), is(1));
+        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(0)),
+                filename,
+                getResourceRetrievalCompletedMessage(resourceData.length()),
+                "complete");
+
+    }
+
+    private void validateNotificationsFailure(CometDClient cometDClient, String filename)
+            throws Exception {
+        List<String> notifications = cometDClient.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
+        assertThat(notifications.size(), is(4));
```

Comment:
While I agree with @coyotesqrl that it's incredibly coupled, I'm more inclined to let it slide because it's a test. If in the future we decide to change it, _then_ it may warrant adding the automated facilities to future-proof it from even more change. 

But I do not foresee the parameters nor circumstances of the test changing. The production logic is getting exercised in the desired fashion.